src/include/duckdb.h

```diff
@@ -2362,6 +2362,15 @@ Fetch the internal arrow schema from the prepared statement.
 */
 DUCKDB_API duckdb_state duckdb_prepared_arrow_schema(duckdb_prepared_statement prepared,
                                                      duckdb_arrow_schema *out_schema);
+
+/*!
+Fetch the internal arrow schema from the query result.
+
+* result: The result object to fetch the schema from.
+* out_schema: The output schema.
+*/
+DUCKDB_API void duckdb_result_arrow_schema(duckdb_result result, duckdb_arrow_schema *out_schema);
```

Comment:
Perhaps it's cleaner to return the `duckdb_arrow_schema` instead, given that there's no other return type?

Reply:
For the sake of consistency and to deal with possible future errors, I could return `duckdb_state` instead. Which do you prefer?